### PR TITLE
TextWidget: adds getFittedText()

### PR DESCRIPTION
**Pull request by poire-z**
Needs a base bump for https://github.com/koreader/koreader-base/pull/1031.
Needed for #5775.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/5776)

### CODE REVIEW

File: frontend/ui/widget/textwidget.lua [hunk 1]
```diff
@@ -201,6 +201,49 @@ function TextWidget:_measureWithXText()
...
+        return self.text, nil
+    end
+    local with_ellipsis = false
+    local start_idx, end_idx = self._shape_start, self._shape_end
+    if self._shape_idx_to_substitute_with_ellipsis then
+        with_ellipsis = true
+        if self.truncate_left then
+            start_idx = start_idx + 1
+        else
+            end_idx = end_idx -1
```
Comment (Frenzie) [inline frontend/ui/widget/textwidget.lua]: -1 or - 1?

File: frontend/ui/widget/textwidget.lua [hunk 2]
```diff
@@ -201,6 +201,49 @@ function TextWidget:_measureWithXText()
     end
 end
 
+function TextWidget:getFittedText()
+    -- Returns the substring of text that fits in self.max_width, without
```
Comment (Frenzie) [inline frontend/ui/widget/textwidget.lua]: Looks like LDoc-type documentation to me, for in http://koreader.rocks/doc/modules/ui.widget.textwidget.html

Like one of these molds:
https://github.com/koreader/koreader/blob/b111ccc2b803fc78425d9791fcc8bb7a9a704e6e/frontend/util.lua#L27-L47
https://github.com/koreader/koreader/blob/b111ccc2b803fc78425d9791fcc8bb7a9a704e6e/frontend/util.lua#L948-L953

Comment (poire-z) [reply frontend/ui/widget/textwidget.lua]: Yeah, but none of the other methods have a ldoc doc. Feels awkward to document such a minor one and not have the other essential ones not documented :) But I don't feel like spending the time needed to document the other ones...

Comment (Frenzie) [reply frontend/ui/widget/textwidget.lua]: Alright, just keep the comment as is then, but I think it'd be clearer above the function. (Also you could change the first line to three dashes to convert it into an LDoc comment in that case, resulting in a smaller diff.)

**Comment by poire-z** [comment]
Note about:
```
-- The substring does not include the ellipsis that could
-- be added when drawn.
```
So, this returns a smaller text string than the one that could fit, as it removes the width needed for the ellipssis. That's fine for #5775 use case.
If one wants the text that fully fit in max_width, without removing the width needed for the ellipsis, one just have to provide `truncate_with_ellipsis = false`, and he will get the expected text.

**Comment by poire-z** [comment]
Just fixed the `not use_xtext` case, after some last minute test.
I'll merge it this evening, even if not validated where it will be used - I'll fix it if needed.

**Pull request merged**